tests/baselines/reference/user/formik.log

```diff
@@ -0,0 +1,20 @@
+Exit Code: 1
+Standard output:
+index.tsx(33,21): error TS2339: Property 'email' does not exist on type 'object'.
```

Comment:
Looks like a result of https://github.com/jaredpalmer/formik/pull/898

Reply:
We should probably update the example as a result. It looks like a lot fewer anys are being inferred.

Reply:
@weswigham,  this is a failure of inference; when I explicitly provide a type argument to `<Formik<...> ...` everything is fine. Maybe this is a result of JSX call resolution? Mind taking a look?

Reply:
Yeah, I'll look into it.

Reply:
I've got a fix up: #27953

Previously JSX did some jank with using apparent types when they shouldn't have which is probably what hid the issue.